src/ray/common/asio/io_context.cc

```diff
@@ -0,0 +1,257 @@
+    const {
+  // We lock the stats table while copying the table into a vector.
+  absl::ReaderMutexLock lock(&mutex_);
+  std::vector<std::pair<std::string, HandlerStats>> stats;
+  stats.reserve(post_handler_stats_.size());
+  std::transform(
+      post_handler_stats_.begin(), post_handler_stats_.end(), std::back_inserter(stats),
+      [](const std::pair<std::string, std::shared_ptr<GuardedHandlerStats>> &p) {
+        return std::make_pair(p.first, to_stats_view(*p.second));
+      });
+  return stats;
+}
+
+std::string to_human_readable(double duration) {
+  static const std::array<std::string, 4> to_unit{"ns", "us", "ms", "s"};
+  size_t idx = std::min(to_unit.size() - 1,
+                        static_cast<size_t>(std::log(duration) / std::log(1000)));
+  double new_duration = duration / std::pow(1000, idx);
+  std::stringstream result;
+  result << std::fixed << std::setprecision(3) << new_duration << " " << to_unit[idx];
+  return result.str();
+}
+
+std::string to_human_readable(int64_t duration) {
+  return to_human_readable(static_cast<double>(duration));
+}
+
+std::string io_context_proxy::StatsString() const {
+  if (!stats_collection_enabled_) {
+    return "Stats collection disabled, turn on asio_event_loop_stats_collection_enabled "
+           "flag to enable event loop stats collection";
+  }
+  std::stringstream result;
+  const auto global_stats = get_global_stats();
+  result << "\nGlobal stats: " << global_stats.cum_count << " total ("
+         << global_stats.curr_count << " active)";
+  result << "\n"
+         << "Queueing time: mean = "
+         << to_human_readable(global_stats.cum_queue_time /
+                              static_cast<double>(global_stats.cum_count))
+         << ", max = " << to_human_readable(global_stats.max_queue_time)
+         << ", min = " << to_human_readable(global_stats.min_queue_time)
+         << ", total = " << to_human_readable(global_stats.cum_queue_time);
+  result << "\n"
+         << "Execution time:  mean = "
+         << to_human_readable(global_stats.cum_execution_time /
+                              static_cast<double>(global_stats.cum_count))
+         << ", max = " << to_human_readable(global_stats.max_execution_time)
+         << ", min = " << to_human_readable(global_stats.min_execution_time)
+         << ", total = " << to_human_readable(global_stats.cum_execution_time);
+  result << "\nHandler stats:";
+  auto stats = get_handler_stats();
+  // Sort stats by cumulative count, outside of the table lock.
+  sort(stats.begin(), stats.end(),
+       [](const std::pair<std::string, HandlerStats> &a,
+          const std::pair<std::string, HandlerStats> &b) {
+         return a.second.cum_count > b.second.cum_count;
+       });
+  for (const auto &entry : stats) {
+    result << "\n\t" << entry.first << ": " << entry.second.cum_count << " total ("
+           << entry.second.curr_count << " active)";
```

Comment:
Can we make this one line per item (remove queueing), and then add total to the this line inline? I think this will greatly improve readability at minimal info loss.

Reply:
Would something like this work?
```
some_service.SomeMethod - 100 total (2 active), CPU time: mean = 500 us, total = 50 ms
```

Reply:
Looks great.